server/controllers/stock/setting/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * The Stock Settings Controller
+ *
+ * This controller is responsible for creating and updating stock-related settings.
+ */
+
+const db = require('../../../lib/db');
+const NotFound = require('../../../lib/errors/NotFound');
+// ??? const BadRequest = require('../../../lib/errors/BadRequest');
+
+// GET /stock/setting
+//
+// Get the current stock settings for the Enterprise
+//    If req.query.enterprise_id is set, it will use that,
+//    otherwise it will look up the entry for Enterprise.id=1
+exports.list = function list(req, res, next) {
+  const enterpriseId = req.query.enterprise_id || '1';
```

Comment:
So, we don't have a great way of handling this case, but most other places in the app just list all the records.  Ironically, because of our inexperience, quite a few of our production enterprise ids are 200, not one.

You can actually get the current enterprise id from `req.session.enterprise.id`.  I propose you do that.

Finally, you should escape this identifier.  Change `WHERE enterprise_id = ${enterpriseId} LIMIT 1;` to `WHERE enterprise_id = ? LIMIT 1;`.  Then pass it in to `db.exec(sql, [enterpriseId])` and that should do the trick.

Reply:
Thanks.  I actually thought of that but could not decide if that was "canonical".

Reply:
I did what you said, but also supported passing the ID in the GET request.  If the parameter is not set, it uses the value from the session.